lib/app/features/auth/views/pages/restore_recovery_keys/restore_recovery_keys_page.dart

```diff
@@ -1,16 +1,71 @@
 // SPDX-License-Identifier: ice License 1.0
 
 import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ice/app/components/card/info_card.dart';
+import 'package:ice/app/components/screen_offset/screen_side_offset.dart';
+import 'package:ice/app/extensions/extensions.dart';
+import 'package:ice/app/features/auth/views/components/auth_scrolled_body/auth_scrolled_body.dart';
 import 'package:ice/app/features/auth/views/components/recovery_keys_input_container/recovery_keys_input_container.dart';
-import 'package:ice/app/router/app_routes.dart';
+import 'package:ice/app/router/components/sheet_content/sheet_content.dart';
+import 'package:ice/app/services/ion_identity_client/ion_identity_client_provider.dart';
+import 'package:ice/generated/assets.gen.dart';
+import 'package:ion_identity_client/ion_client.dart';
 
-class RestoreRecoveryKeysPage extends StatelessWidget {
+class RestoreRecoveryKeysPage extends HookConsumerWidget {
   const RestoreRecoveryKeysPage({super.key});
 
   @override
-  Widget build(BuildContext context) {
-    return RecoveryKeysInputContainer(
-      onContinuePressed: () => TwoFaOptionsRoute().push<void>(context),
+  Widget build(BuildContext context, WidgetRef ref) {
+    final ionClient = ref.watch(ionApiClientProvider.future);
+
+    final recoverUserResult = useState<RecoverUserResult?>(null);
+
+    if (recoverUserResult.value == null) {
+      return FutureBuilder<IonApiClient>(
```

Comment:
Is this `FutureBuilder` temporary solution?

Reply:
Done. Update the code to remove `FutureBuilder`.